pettingzoo/utils/conversions.py

```diff
@@ -335,7 +349,7 @@ def step(self, action: ActionType):
 
             self.agents = self.env.agents + [
                 agent
-                for agent in sorted(self._observations.keys())
+                for agent in sorted(self._observations.keys(), key=lambda x: hash(x))
```

Comment:
I'm assuming this is to be able to handle non-strings? Does hashing produce any undesirable side effects though? I would imagine you would probably want it to sort in a specific order otherwise you wouldn't call sorted (I'm not familiar with the previous code here)

Reply:
Yes, the reason is in order to handle non-strings.

The reason we call sorted here, at least as far as I know, is so that the `agents` list has a deterministic ordering, since otherwise a dictionary makes no guarantees as to the order of its items.

You actually raise a good point here, since `hash` values are not consistent between different runs of the python interpreter. 

I should use the `str` function to convert it to a string for comparison instead here.

Reply:
Done now.